ddtrace/llmobs/_integrations/base_stream_handler.py

```diff
@@ -0,0 +1,231 @@
+        
+        Args:
+            exception: The exception that occurred
+        """
+        if self.primary_span:
+            self.primary_span.set_exc_info(*sys.exc_info())
+
+    @abstractmethod
+    def finalize_stream(self, exception=None):
+        """
+        Finalize the stream and complete all spans.
+        
+        This method is called when the stream ends (successfully or with error).
+        Implementations should:
+        1. Process accumulated chunks into final response
+        2. Set appropriate span tags
+        3. Finish all spans
+        """
+        pass
+
+
+class StreamHandler(BaseStreamHandler):
+    @abstractmethod
+    def process_chunk(self, chunk, iterator=None):
+        """
+        Process a single chunk from the stream.
+        
+        This method is called for each chunk as it's received.
+        Implementations should extract and store relevant data.
+        
+        Args:
+            chunk: The chunk object from the stream
+            iterator: The sync iterator object from the stream
+        """
+        pass
+
+
+class AsyncStreamHandler(BaseStreamHandler):
+    @abstractmethod
+    async def process_chunk(self, chunk, iterator=None):
+        """
+        Process a single chunk from the stream.
+        
+        This method is called for each chunk as it's received.
+        Implementations should extract and store relevant data.
+        
+        Args:
+            chunk: The chunk object from the stream
+            iterator: The async iterator object from the stream
+        """
+        pass
+
+
+class _ClassTracedStream(wrapt.ObjectProxy):
+    def __init__(self, wrapped_stream, handler: StreamHandler, on_stream_created=None):
+        super().__init__(wrapped_stream)
+        self._handler = handler
+        self._on_stream_created = on_stream_created
+        self._exception = None
+        if hasattr(self.__wrapped__, '__next__') or not hasattr(self.__wrapped__, '__iter__'):
+            self._stream_iter = self.__wrapped__
+        else:
+            self._stream_iter = iter(self.__wrapped__)
+    
+    def __iter__(self):
+        try:
+            for chunk in self._stream_iter:
+                self._handler.process_chunk(chunk, self._stream_iter)
+                yield chunk
+        except Exception as e:
+            self._exception = e
+            self._handler.handle_exception(e)
+            raise
+        finally:
+            self._handler.finalize_stream(self._exception)
+    
+    def __enter__(self):
+        if hasattr(self.__wrapped__, '__enter__'):
+            result = self.__wrapped__.__enter__()
+            # update iterator in case we are wrapping a stream manager
+            if result is not self.__wrapped__:
+                self._stream_iter = iter(result) if not hasattr(result, '__next__') else result
+                new_stream = _ClassTracedStream(result, self._handler, self._on_stream_created)
```

Comment:
```suggestion
                traced_stream = _ClassTracedStream(result, self._handler, self._on_stream_created)
```
Does this make more sense as the var name?

Reply:
yeah this makes sense to me!